sycl/include/CL/sycl/queue.hpp

```diff
@@ -241,7 +351,25 @@ class __SYCL_EXPORT queue {
   event submit(T CGF, queue &SecondaryQueue _CODELOCPARAM(&CodeLoc)) {
     _CODELOCARG(&CodeLoc);
 
-    return submit_impl(CGF, SecondaryQueue, CodeLoc);
+    event Event;
+
+#if !defined(SYCL_DISABLE_FALLBACK_ASSERT) && !defined(__NVPTX__)
+    bool IsKernel = false;
+    Event = submit_impl(CGF, IsKernel, SecondaryQueue, CodeLoc);
```

Comment:
Can we extend `event_impl` to store CGType and based on this information understand whether we are submitting a kernel or not?
Or can we change submit_impl interface to take a lambda which takes all necessary things like event and handler and enqueues assert related tasks if needed.

Reply:
Added postprocessing lambda